### ISSUE TITLE
dsnt calculate numerical coord in heatmaps

### ISSUE THREAD

**Issue by zccyman** [kornia/kornia#274]
I think this dsnt is similar to coordinate calculation of (PoseFix)[https://arxiv.org/pdf/1812.03595.pdf].Is it possible to substitute posefix method for dsnt. In my opinion, I use posefix method more convenient  
than dsnt,  no need to convert to image size from small digits in pose estimation.

```
#(Numerical Coordinate Regression with Convolutional Neural Networks)[https://arxiv.org/pdf/1801.07372.pdf]     
def get_coord_use_dsntnn(self, outputs):
        h, w = outputs.shape[2], outputs.shape[3]

        heatmaps = dsntnn.flat_softmax(outputs)
        coords = dsntnn.dsnt(heatmaps)  # [b, c, (x, y)]
        coords += 1.0

        coords[:, :, 0] *= w
        coords[:, :, 1] *= h
        coords = (coords - 1.0) / 2.0

        return coords

#(PoseFix: Model-agnostic General Human Pose Refinement Network)[https://arxiv.org/pdf/1812.03595.pdf]
def get_coord_use_posefix(self, outputs):
        batch_size, keypoints_num, height, width = outputs.shape
        grid_x = torch.arange(0, width).expand((height, width)) + 1
        grid_y = torch.arange(0, height).reshape(
            height, 1).expand((height, width)) + 1
        grid_x = (
            grid_x.expand_as(outputs).cuda(
                device=torch.cuda.current_device()).float()
        )
        grid_y = (
            grid_y.expand_as(outputs).cuda(
                device=torch.cuda.current_device()).float()
        )

        outputs = F.softmax(
            torch.reshape(outputs, (batch_size, keypoints_num, -1)), dim=2
        )
        outputs = torch.reshape(
            outputs, (batch_size, keypoints_num, height, width))

        outputs_x = outputs.mul(grid_x)
        outputs_y = outputs.mul(grid_y)

        outputs_x = torch.reshape(
            outputs_x, (batch_size, keypoints_num, -1))
        outputs_y = torch.reshape(
            outputs_y, (batch_size, keypoints_num, -1))

        outputs_x = torch.sum(outputs_x, dim=2).unsqueeze(dim=2)
        outputs_y = torch.sum(outputs_y, dim=2).unsqueeze(dim=2)

        coord_pred = torch.cat([outputs_x, outputs_y], dim=2)
        coord_pred = coord_pred - 1

        return coord_pred

**Comment by edgarriba** [comment]
@anibali could you take a look at it ?

**Comment by anibali** [comment]
@zccyman The PoseFix paper literally says that they use soft-argmax, which is the same as DSNT. I don't understand what you're asking for. If all you want is to use pixel coordinates instead of normalised coordinates, you can try out the `normalized_coordinates` flag for `spatial_softargmax_2d`:

https://github.com/kornia/kornia/blob/0a3cbb02850ac78059e0615da93144b5a64d3330/kornia/geometry/dsnt.py#L54-L85
